programs/agora-court/src/instructions/vote.rs

```diff
@@ -1,7 +1,70 @@
-/*
-    1. Verify dispute is in progress then call `close_dispute` to ensure that
-       it fails
-    2. Check that voter has rep >= `dispute.config.rep_required`
-    3. Either update `case.votes` with voter PK or create PDA OR increment 
-       `dispute.abstained_votes`
-*/
\ No newline at end of file
+use crate::{constant::USER_MAX_DISPUTES, error::InputError, state::*};
+use anchor_lang::prelude::*;
+
+#[derive(Accounts)]
+#[instruction(dispute_id: u64, user_case: Pubkey)]
+pub struct Vote<'info> {
+    #[account(
+        mut,
+        seeds = [b"case".as_ref(), court.key().as_ref(), dispute.key().as_ref(), user_case.as_ref()],
+        bump = case.bump,
+   )]
+    pub case: Account<'info, Case>,
+
+    #[account(
+        mut,
+        seeds = [b"reputation".as_ref(), court_authority.key().as_ref(), payer.key().as_ref()],
+        bump = reputation.bump,
+        constraint = !reputation.in_dispute(dispute_id)
+                    @ InputError::UserAlreadyVoted,
+
+        constraint = !reputation.has_unclaimed_disputes()
+                    @ InputError::UserHasUnclaimedDisputes,
+
+        constraint = reputation.claim_queue.len() < USER_MAX_DISPUTES
+                    @ InputError::UserMaxDisputesReached,
+
+        constraint = reputation.reputation >= dispute.config.rep_required
+                    @ InputError::UserDoesNotHaveEnoughReputation,
+    )]
+    pub reputation: Account<'info, Reputation>,
+
+    #[account(
+        mut,
+        seeds = [b"dispute".as_ref(), court.key().as_ref(), u64::to_ne_bytes(dispute_id).as_ref()],
+        bump = dispute.bump,
+        constraint = dispute.status == DisputeStatus::Voting
+                    @ InputError::DisputeNotVotable,
```

Comment:
I purposely left out any mention of `close_dispute` in `vote` because I think it is clearer to simply enforce here that dispute must be votable and leave `close_dispute` as a separate instruction. I think ideally we avoid having multiple outcomes instructions, like having `vote` either vote or close a dispute.

Reply:
i agree, i think its cleaner the way you did it too